src/WebJobs.Script.WebHost/Metrics/FlexConsumptionMetricsPublisher.cs

```diff
@@ -203,6 +211,16 @@ private bool PrepareDirectoryForFile()
             return files.Count < _options.MaxFileCount;
         }
 
+        private HostMetrics PrepareHostMetrics()
```

Comment:
Might be better for MetricsCollector.GetHostMetrics to do all this logic internally and just return a HostMetrics instance. Might be GetHostMetricsOrNull and return null if no metrics to report.